# Bootcamp Arquiteto de Dados
# Trabalho Prático 2

por: Inácio Miura Junior


In [ ]:
import pandas as pd
import numpy as np

from datetime import datetime, date #biblioteca pra converter as datas em idade

Importando e olhando o dataset

In [ ]:
add_base = '/content/base.csv'
df_base= pd.read_csv(add_base, sep= ';', encoding='latin-1')
df_base.head(3)

cod_pessoa data_coleta     genero data_nascimento animal_estimacao   clima  \
0    10141743  2020-01-01  Masculino      1981-04-25            peixe    frio   
1    10141860  2020-01-01   Feminino      1966-01-19            peixe  quente   
2    10142548  2020-01-01  Masculino      1973-08-02        tartaruga    frio   

  bebida_favorita           hobbies  
0         Cerveja  Praticar esporte  
1            Café       Assistir TV  
2            Café        Ler livros

In [ ]:
pd.set_option('display.float_format', lambda x: '%.2f' % x) #Definindo o formato do output do pandas

df_base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2127 entries, 0 to 2126
Data columns (total 8 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   cod_pessoa        2127 non-null   int64 
 1   data_coleta       2127 non-null   object
 2   genero            2127 non-null   object
 3   data_nascimento   2127 non-null   object
 4   animal_estimacao  2127 non-null   object
 5   clima             2127 non-null   object
 6   bebida_favorita   2127 non-null   object
 7   hobbies           2127 non-null   object
dtypes: int64(1), object(7)
memory usage: 133.1+ KB


Substituindo o tipo de variável das Datas para ficar mais fácil de trabalhar

In [ ]:
df_base['data_nascimento'] = pd.to_datetime(df_base['data_nascimento'])
df_base['data_coleta'] = pd.to_datetime(df_base['data_coleta'])

## Verificando dados ausentes:

In [ ]:
df_base.isna().sum()

cod_pessoa          0
data_coleta         0
genero              0
data_nascimento     0
animal_estimacao    0
clima               0
bebida_favorita     0
hobbies             0
dtype: int64

In [ ]:
df_base['cod_pessoa'].value_counts().count() #contei o número de entradas únicas para ter certeza que não há registros duplicados

2127

## Q1: Quantas pessoas do genero masculino gostam de peixe?

In [ ]:
df_base.where((df_base.animal_estimacao == 'peixe') & (df_base.genero == 'Masculino')).count()


cod_pessoa          274
data_coleta         274
genero              274
data_nascimento     274
animal_estimacao    274
clima               274
bebida_favorita     274
hobbies             274
dtype: int64

## Q2: Qual a média de idade dos mulheres que gostam de chá e clima frio?

In [ ]:
df_base['bebida_favorita'].value_counts() #Verificando como está escrito no DF

Cerveja         388
Café            383
refrigerante    350
Chá             348
Vinho           333
Água            325
Name: bebida_favorita, dtype: int64

In [ ]:
df_base['clima'].value_counts() #Verificando como está escrito no DF

frio        723
moderado    711
quente      693
Name: clima, dtype: int64

In [ ]:
now = pd.Timestamp('now')   #definindo uma variável de comparação para idade
df_base['idade'] = (now - df_base['data_nascimento']).astype('timedelta64[Y]')
df_base.head(3)

cod_pessoa data_coleta     genero data_nascimento animal_estimacao   clima  \
0    10141743  2020-01-01  Masculino      1981-04-25            peixe    frio   
1    10141860  2020-01-01   Feminino      1966-01-19            peixe  quente   
2    10142548  2020-01-01  Masculino      1973-08-02        tartaruga    frio   

  bebida_favorita           hobbies  idade  
0         Cerveja  Praticar esporte  41.00  
1            Café       Assistir TV  56.00  
2            Café        Ler livros  48.00

In [ ]:
filtro1='Feminino'
filtro2='Chá'
filtro3='frio'

df_base.query('genero == @filtro1').query('bebida_favorita == @filtro2').query('clima == @filtro3').describe()


cod_pessoa  idade
count       55.00  55.00
mean  10142233.89  49.60
std        734.24  14.11
min   10141516.00  22.00
25%   10141618.50  37.00
50%   10141909.00  55.00
75%   10142877.50  60.00
max   10143600.00  72.00

## Q3: Qual o hobbie de menor preferência entre as mulheres?

In [ ]:
filtro1='Feminino'

df_base.query('genero == @filtro1')['hobbies'].value_counts()

Escutar música        103
Escrever               94
Praticar esporte       93
Dormir                 92
Assistir TV            90
Pintar quadros         89
Aprender algo novo     88
Ler livros             80
Name: hobbies, dtype: int64

## Q4: Qual o hobbie de maior preferência entre as mulheres que gostam de  cachorro?

In [ ]:
df_base['animal_estimacao'].value_counts() #Verificando como está escrito no DF

gato         864
peixe        432
tartaruga    417
cachorro     414
Name: animal_estimacao, dtype: int64

In [ ]:
filtro1='Feminino'
filtro2='cachorro'
df_base.query('genero == @filtro1').query('animal_estimacao == @filtro2')['hobbies'].value_counts()

Praticar esporte      22
Dormir                21
Ler livros            18
Escutar música        18
Pintar quadros        18
Assistir TV           16
Escrever              15
Aprender algo novo    15
Name: hobbies, dtype: int64

## Q5: Qual a bebida favorita entre os mulheres e homens? respectivamente?

In [ ]:
filtro1= 'Cerveja'
df_base.groupby('genero')['bebida_favorita'].value_counts()

genero     bebida_favorita
Feminino   Cerveja            138
           Café               125
           Chá                125
           refrigerante       124
           Vinho              114
           Água               103
Masculino  Café               258
           Cerveja            250
           refrigerante       226
           Chá                223
           Água               222
           Vinho              219
Name: bebida_favorita, dtype: int64

## Q6: Qual é a média de idade das pessoas que tem como hobbie assistir TV?

In [ ]:
df_base['hobbies'].value_counts() #Verificando como está escrito no DF

Escrever              276
Dormir                276
Aprender algo novo    271
Praticar esporte      268
Assistir TV           263
Pintar quadros        260
Ler livros            257
Escutar música        256
Name: hobbies, dtype: int64

In [ ]:
filtro1='Assistir TV'

df_base.query('hobbies == @filtro1').describe()

cod_pessoa  idade
count      263.00 263.00
mean  10142530.99  45.92
std        604.06  14.35
min   10141501.00  22.00
25%   10141996.00  33.00
50%   10142550.00  46.00
75%   10143010.00  58.00
max   10143616.00  72.00

## Q7:Quantas pessoas tem como Hobbie como pintar quadros e qual a sua média de idade respectivamente?

In [ ]:
df_base['hobbies'].value_counts() #Verificando como está escrito no DF

Escrever              276
Dormir                276
Aprender algo novo    271
Praticar esporte      268
Assistir TV           263
Pintar quadros        260
Ler livros            257
Escutar música        256
Name: hobbies, dtype: int64

In [ ]:
filtro1='Pintar quadros'

df_base.query('hobbies == @filtro1').describe()

cod_pessoa  idade
count      260.00 260.00
mean  10142597.97  46.16
std        592.25  14.76
min   10141507.00  22.00
25%   10142099.00  33.00
50%   10142653.00  45.00
75%   10143125.50  59.00
max   10143622.00  72.00

## A questão 8 é discursiva

## Q9: Quantas pessoas de modo geral gostam de cachorro?

In [ ]:
df_base['animal_estimacao'].value_counts()

gato         864
peixe        432
tartaruga    417
cachorro     414
Name: animal_estimacao, dtype: int64

## Q10: Qual o segundo animal de estimação preferido entre os homens?

In [ ]:
filtro1='Masculino'

df_base.query('genero == @filtro1')['animal_estimacao'].value_counts()

gato         585
peixe        274
cachorro     271
tartaruga    268
Name: animal_estimacao, dtype: int64

## Q11: Quantas pessoas que gostam do clima quente?

In [ ]:
df_base['clima'].value_counts() 

frio        723
moderado    711
quente      693
Name: clima, dtype: int64

Encontrei 693 mas só tem 692 entre as alternativas. Está certo?